### DISCUSSION TITLE
Render Order for Scatterplot Points

### DISCUSSION THREAD

**Discussion by ilan-gold** [visgl/deck.gl#6138] [Q&A] [answered]
Is there a way to tell the `ScatterplotLayer` what order to render points in?  Or to give priority to a set of points?

**Comment by ibgreen** [comment, marked as answer]
The beauty of GPU rendering is that it happens in parallel, so ordering would typically be handled by other mechanisms such as small z offsets etc. There is often a way, but it tends to depend on what you want to achieve.

**Reply by ilan-gold** [reply]
Hmmm, ok.  For some reason I had it in my mind that each point got its own `draw` call but that isn't the case.  The z-offset sounds interesting, if it would affect the 2D rendering.  Beyond that we thought maybe about two different layers, one on top of the other.  I'll post here what we land on.

**Reply by ilan-gold** [reply]
We're trying to get a selected subset of scatter plot points in one color to render "on top" of another unselected group.  So if there is ever overlap between two points  because they have some nonzero radius, the selected point will always render on top of the unselected point.

**Reply by ibgreen** [reply]
If you can determine the selection status of a point in the shader (either by adding a new attribute or e.g. comparing the point color), you could apply a tiny z offset just to those points.

**Reply by ilan-gold** [reply]
@ibgreen That worked great.  We're messing around with opacity so that created its own headaches where selected i.e yellow points that have low opacity blend poorly with the background (the black dots, I think) since they are in the foreground when they are not near other yellow points
<img width="222" alt="Screen Shot 2021-08-27 at 12 36 43 PM" src="https://user-images.githubusercontent.com/43999641/131160378-a82ad6fd-d672-4583-9eba-0d2d367c4d1c.png">
...a bit of wack-a-mole
